examples/mfa/webauthn/static/index.html

```diff
@@ -0,0 +1,165 @@
+                }
+            };
+            
+            if (credentials.response.userHandle) {
+                cred.response.userHandle = base64UrlEncode(credentials.response.userHandle);
+            }
+
+            // Send credentials back to server to save
+            let res = await fetch('/register/finish', {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify(credentialsForServer)
+            });
+
+            if (!res.ok) {
+                throw new Error(`HTTP error! status: ${res.status}`);
+            }
+
+            console.log("Registration complete!");
+        }
+
+        async function login() {
+            let response = await fetch('/login/begin', {
+                method: 'GET'
+            });
+
+            if (!response.ok) {
+                throw new Error(`HTTP error! status: ${response.status}`);
+            }
+
+            let data = await response.json();
+            data.publicKey.challenge = base64UrlDecode(data.publicKey.challenge)
+            data.publicKey.allowCredentials.forEach(function (listItem) {
+                listItem.id = base64UrlDecode(listItem.id)
+            });
+
+            let assertion = await navigator.credentials.get(data);
+
+            // Send credentials back to server to save
+            let res = await fetch('/login/finish', {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify({
+                    id: assertion.id,
+                    rawId: base64UrlEncode(assertion.rawId),
+                    type: assertion.type,
+                    response: {
+                        authenticatorData: base64UrlEncode(assertion.response.authenticatorData),
+                        clientDataJSON: base64UrlEncode(assertion.response.clientDataJSON),
+                        signature: base64UrlEncode(assertion.response.signature),
+                        userHandle: base64UrlEncode(assertion.response.userHandle),
+                    }
+                })
+            });
+
+            if (!res.ok) {
+                throw new Error(`HTTP error! status: ${res.status}`);
+            }
+
+            console.log("Login complete!");
+        }
+
+        document.addEventListener('DOMContentLoaded', async (event) => {
+            const statusDiv = document.getElementById('status');
+
+            const response = await fetch('/check-registration', {
+                method: 'GET',
+            });
+
+            let data = response.json();
+
+            if (!data.isRegistered) {
+                try {
+                    await register();
+                } catch(error) {
+                    statusDiv.textContent = `Error during registration: ${error}`;
```

Comment:
Any danger of XSS attack in error? Is error guarantied to be escaped?

Reply:
I hear what you're saying about XSRF and XSS, but I don't think those should be a concern in an example which isn't meant for practical application and whose preventative details wouldn't have an impact on opk

Reply:
We should avoid XSS in example javascript in case someone uses them and to establish patterns of security. If something is unsafe in an example we should leave a comment.

I did some digging and `.textContent` naturally protects against XSS and can safely be used.

> RULE #6 - Populate the DOM using safe JavaScript functions or properties The most fundamental safe way to populate the DOM with untrusted data is to use the safe assignment property textContent.

[OWASP: DOM based XSS Prevention Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/DOM_based_XSS_Prevention_Cheat_Sheet.html#rule-6-populate-the-dom-using-safe-javascript-functions-or-properties)

I consider my comment here addressed and I am convinced this is secure.